src/main/scala/Ctx.scala

```diff
@@ -1,45 +1,47 @@
 import akka.NotUsed
-import akka.util.Timeout
-import schema.MutationError
 import akka.actor.ActorRef
-import generic.Event
-import generic.MemoryEventStore._
-import generic.View.{Get, GetMany}
 import akka.pattern.ask
 import akka.stream.OverflowStrategy
 import akka.stream.scaladsl.Source
-import org.reactivestreams.Publisher
+import akka.util.Timeout
+import generic.Event
+import generic.PersistentEventStore._
+import generic.View.{Get, GetMany}
+import schema.MutationError
 
+import scala.concurrent.duration._
 import scala.concurrent.{ExecutionContext, Future}
+import scala.language.postfixOps
 
 case class Ctx(
   authors: ActorRef,
   articles: ActorRef,
   eventStore: ActorRef,
-  eventStorePublisher: Publisher[Event],
+  events: Source[Event, NotUsed],
   ec: ExecutionContext,
   to: Timeout
 ) extends Mutation {
   implicit def executionContext = ec
   implicit def timeout = to
 
+  // TODO why buffer?
   lazy val eventStream: Source[Event, NotUsed] =
-    Source.fromPublisher(eventStorePublisher).buffer(100, OverflowStrategy.fail)
+    events.buffer(100, OverflowStrategy.fail).backpressureTimeout(3 seconds)
```

Comment:
It is for slow subscribers. If client connection is too slow an can't keep up with an even stream, it will fail and will be disconnected. In this case I don't want to propagate back-pressure back to main event bus since one slow client will then slow down the whole system.